src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs

```diff
@@ -1,22 +1,39 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
 
 namespace Microsoft.CodeAnalysis.Operations
 {
     /// <summary>
     /// PROTOTYPE(dataflow): Add documentation
     /// </summary>
-    public sealed class ControlFlowGraph
+    public sealed partial class ControlFlowGraph
     {
-        internal ControlFlowGraph(ImmutableArray<BasicBlock> blocks)
+        internal ControlFlowGraph(ImmutableArray<BasicBlock> blocks, Region root)
         {
+            Debug.Assert(!blocks.IsDefault);
```

Comment:
Should be IsDefaultOrEmpty, or perhaps add one for IsEmpty, as it will null ref below if it is.

Reply:
> Should be IsDefaultOrEmpty, or perhaps add one for IsEmpty, as it will null ref below if it is.

I think the following asserts are sufficient to cover this expectation. It is not my goal to avoid a null ref on an invalid input.  

---
In reply to: [167649617](https://github.com/dotnet/roslyn/pull/24770#discussion_r167649617) [](ancestors = 167649617)

Reply:
Then why have the is IsDefault at all? The cases below will cover that as well.

---
In reply to: [167651458](https://github.com/dotnet/roslyn/pull/24770#discussion_r167651458) [](ancestors = 167651458,167649617)

Reply:
> Then why have the is IsDefault at all?

Is there any harm in having it?

---
In reply to: [167651658](https://github.com/dotnet/roslyn/pull/24770#discussion_r167651658) [](ancestors = 167651658,167651458,167649617)

Reply:
The same amount of harm in having an explicit IsEmpty :)

---
In reply to: [167652102](https://github.com/dotnet/roslyn/pull/24770#discussion_r167652102) [](ancestors = 167652102,167651658,167651458,167649617)

Reply:
My point is that an assert for empty is not required. And, therefore, the suggested change is not required.

---
In reply to: [167652440](https://github.com/dotnet/roslyn/pull/24770#discussion_r167652440) [](ancestors = 167652440,167652102,167651658,167651458,167649617)